File: backend/Tests/AnarchyChess.Api.TestInfrastructure/Fakes/VoteOptionFaker.cs
using AnarchyChess.Api.Vote.Entities;
using AnarchyChess.Api.Vote.Models;
using Bogus;

namespace AnarchyChess.Api.TestInfrastructure.Fakes;

public class VoteOptionFaker : Faker<VoteOption>
{
    public VoteOptionFaker()
    {
        StrictMode(true);
        RuleFor(x => x.Key, f => (VoteOptionKey)f.Random.Guid().ToString());
        RuleFor(x => x.Name, f => f.Lorem.Sentence(wordCount: 3));
        RuleFor(x => x.Description, f => f.Lorem.Paragraph());
    }
}

File: frontend/src/lib/testUtils/fakers/voteOptionFaker.ts
import { VoteOption } from "@/lib/apiClient";
import { faker } from "@faker-js/faker";

export function createFakeVoteOption(
    overrides?: Partial<VoteOption>,
): VoteOption {
    return {
        optionKey: faker.string.uuid(),
        name: faker.lorem.words(),
        description: faker.lorem.paragraph(),
        ...overrides,
    };
}
